File: RepoTransBench/target_projects/Python/Java/whonore_Coqtail/src/test/java/com/whonore/coqtail/original/TestParse.java
package com.whonore.coqtail.original;

import org.junit.jupiter.api.Test;
import com.whonore.coqtail.parse.Parse;
import static org.junit.jupiter.api.Assertions.*;

public class TestParse {

    @Test
    void testParseIdentifierAlpha() {
        assertTrue(Parse.isIdent("KappaZetaXYZ"));
    }

    @Test
    void testParseIdentifierMixed() {
        assertTrue(Parse.isIdent("T2X9P"));
    }

    @Test
    void testParseNonIdentifierNumeric() {
        assertFalse(Parse.isIdent("10MainVar"));
    }

    @Test
    void testParseSplitLineColon() {
        String text = "Theorem Power: forall n, n ^ 2 >= 0.";
        String[] result = Parse.splitLine(text);
        assertTrue(result instanceof String[]);
        assertTrue(result[0].contains("Theorem"));
    }

    @Test
    void testParseFindNameTheorem() {
        String text = "Theorem my_power: forall n, n ^ 2 >= 0.";
        String name = Parse.findName(text);
        assertTrue(name == null || name instanceof String);
    }
}

File: RepoTransBench/target_projects/Python/Rust/whonore_Coqtail/tests/original/test_parse.rs
use coqtail::parse::*;

#[test]
fn test_is_ident() {
    assert!(is_ident("KappaZetaXYZ"));
    assert!(is_ident("T2X9P"));
    assert!(!is_ident("10MainVar"));
    assert!(!is_ident("_bad"));
    assert!(!is_ident(""));
}

#[test]
fn test_split_line() {
    let text = "Theorem Power: forall n, n ^ 2 >= 0.";
    let result = split_line(text);
    assert_eq!(result.0, "Theorem Power");
    assert!(result.1.contains("forall"));
}

#[test]
fn test_find_name() {
    let text = "Theorem my_power: forall n, n ^ 2 >= 0.";
    let name = find_name(text);
    assert_eq!(name.unwrap(), "my_power");
}